### PR TITLE
Experimental

### PR DESCRIPTION
**Pull request by silkspace**

### CODE REVIEW

File: graphistry/Plottable.py [hunk 1]
```diff
@@ -75,6 +75,7 @@ class Plottable(object):
     _entity_to_index : dict
     _index_to_entity : dict
 
+    DGL_graph: Optional[Any]
```
Comment (lmeyerov) [inline graphistry/Plottable.py:78]: rename to _dgl_graph ?

File: graphistry/dgl_utils.py [hunk 2]
```diff
@@ -18,27 +19,26 @@
     XSymbolic,
     YSymbolic,
     resolve_X,
-    resolve_y
+    resolve_y,
 )
 from .util import setup_logger
 
-logger = setup_logger(__name__)
+logger = setup_logger(name=__name__, verbose=config.VERBOSE)
```
Comment (lmeyerov) [inline graphistry/dgl_utils.py:26]: let setup_logger handle defaults

File: graphistry/dgl_utils.py [hunk 3]
```diff
@@ -18,27 +19,26 @@
...
+
 # #########################################################################################
 #
 #  Torch helpers
 #
 # #########################################################################################
 
 
-def convert_to_torch(X_enc: pd.DataFrame, y_enc: Optional[pd.DataFrame]):
+def convert_to_torch(X_enc: pd.DataFrame, y_enc: Optional[pd.DataFrame]): # type: ignore
```
Comment (lmeyerov) [inline graphistry/dgl_utils.py:41]: suspicious

File: graphistry/dgl_utils.py [hunk 4]
```diff
@@ -295,85 +306,137 @@ def _convert_edgeDF_to_DGL(self, node_column: Optional[str] = None, weight_colum
...
+
+        X_enc, y_enc, res = res._featurize_or_get_nodes_dataframe_if_X_is_None(
+            X=X,
+            y=y,
+            use_scaler=use_scaler,
+            feature_engine=resolve_feature_engine(feature_engine),
+        )
+
+        logger.debug(
+            f"=*=*=Encoded Node shapes are data: {X_enc.shape}, target: {y_enc.shape}"
```
Comment (lmeyerov) [inline graphistry/dgl_utils.py:331]: do not use `f'{...}` during logging, should be `logger.debug("some string: %s", some_var)` : that way the string interpolation gets skipped when the log mode is off

File: graphistry/feature_utils.py [hunk 5]
```diff
@@ -9,7 +9,7 @@
 from .PlotterBase import WeakValueDictionary, Plottable
 from .util import setup_logger, check_set_memoize
 
-logger = setup_logger(__name__)
+logger = setup_logger(name=__name__, verbose=config.VERBOSE)
```
Comment (lmeyerov) [inline graphistry/feature_utils.py:12]: let setup_logger handle defaults

File: graphistry/feature_utils.py [hunk 6]
```diff
@@ -548,36 +548,76 @@ def impute_and_scale_df(
...
-            # assuming they are unique across cols
+        res = df[text_cols[0]].astype(str)
+        if len(text_cols) > 1:
+            for col in text_cols[1:]:
+                res += ' ' + df[col].astype(str)
+        print(res)
+        print(type(res))
+        if use_ngrams:
+            model = get_text_preprocessor(ngram_range, max_df, min_df)
+            logger.debug(f"-Calculating Tfidf Vectorizer with {ngram_range}-ngrams for column(s) `{text_cols}`")
```
Comment (lmeyerov) [inline graphistry/feature_utils.py:588]: `f'{var'}` -> `logger.debug("asdf %s", var)`

File: graphistry/feature_utils.py [hunk 7]
```diff
@@ -548,36 +548,76 @@ def impute_and_scale_df(
...
+            if scipy.sparse.issparse(emb):
+                emb = emb.toarray()
             embeddings = np.c_[embeddings, emb]
-        logger.debug(
-            f"Encoded Textual data at {len(df)/(len(text_cols)*(time()-t)/60):.2f} rows per column minute"
-        )
-
-    return embeddings, text_cols, columns
+            logger.debug(
+                f"Encoded Textual using Ngrams {len(df)/(len(text_cols)*(time()-t)/60):.2f} rows per minute"
```
Comment (lmeyerov) [inline graphistry/feature_utils.py:596]: another

File: graphistry/feature_utils.py [hunk 8]
```diff
@@ -548,36 +548,76 @@ def impute_and_scale_df(
...
+            )
+        else:
+            model_name = os.path.split(model_name)[-1]
+            model = SentenceTransformer(f"{model_name}")
+            emb = model.encode(res.values)
+            embeddings = np.c_[embeddings, emb]
+            logger.debug(
+                f"Encoded Textual using Embedding {model_name}  at {len(df)/(len(text_cols)*(time()-t)/60):.2f} rows per minute"
```
Comment (lmeyerov) [inline graphistry/feature_utils.py:604]: another

File: graphistry/pygraphistry.py [hunk 9]
```diff
@@ -1572,13 +1577,23 @@ def gsql(query, bindings=None, dry_run=False):
                      tg = graphistry.tigergraph()
                      tg.gsql(\"\"\"
                      INTERPRET QUERY () FOR GRAPH Storage {
+<<<<<<< HEAD
```
Comment (lmeyerov) [inline graphistry/pygraphistry.py:1580]: @silkspace bad merge?

File: graphistry/tests/test_feature_utils.py [hunk 10]
```diff
@@ -1,12 +1,18 @@
 # python -m unittest
+import datetime as dt
+import graphistry
+import logging
```
Comment (lmeyerov) [inline graphistry/tests/test_feature_utils.py:4]: @silkspace are you using some tool that is breaking your formatting, causing these spurious (and convention-breaking) changes?

**Comment by lmeyerov** [comment]
@silkspace can we remove any of your branches/PRs? seems to be accumulating

**Comment by lmeyerov** [comment]
@silkspace can we delete this branch / pr, i think it's already merged?

File: graphistry/Plottable.py (hunk 1 above)
Comment (silkspace) [reply graphistry/Plottable.py:78]: yes -- I'll do this on current `cleanup: dbscan and infer_graph` branch

**Comment by silkspace** [comment]
Old stale PR

**Pull request closed without merging**